src/cors.ts

```diff
@@ -0,0 +1,175 @@
+  return typeof s === 'string' || s instanceof String
+}
+
+function isOriginAllowed(
+  origin: string,
+  allowedOrigin: (string | RegExp)[] | RegExp | string | boolean | CustomOrigin
+) {
+  if (Array.isArray(allowedOrigin)) {
+    for (let i = 0; i < allowedOrigin.length; ++i) {
+      if (isOriginAllowed(origin, allowedOrigin[i])) {
+        return true
+      }
+    }
+    return false
+  } else if (isString(allowedOrigin)) {
+    return origin === allowedOrigin
+  } else if (allowedOrigin instanceof RegExp) {
+    return allowedOrigin.test(origin)
+  } else if (allowedOrigin instanceof Function) {
+    return allowedOrigin(origin)
+  } else {
+    return !!allowedOrigin
+  }
+}
+
+function configureOrigin(options: CorsOptions, req: Request, headers: Headers) {
+  const requestOrigin = req.headers.get('origin')
+  let isAllowed: boolean = false
+
+  if (!options.origin || options.origin === '*') {
+    // allow any origin
+    headers.set('Access-Control-Allow-Origin', '*')
+  } else if (isString(options.origin)) {
+    // fixed origin
+    headers.set('Access-Control-Allow-Origin', options.origin)
+    headers.set('Vary', 'Origin')
+  } else if (requestOrigin && options.origin) {
+    isAllowed = isOriginAllowed(requestOrigin, options.origin)
+    // reflect origin
+    headers.set('Access-Control-Allow-Origin', isAllowed ? requestOrigin : 'false')
+    headers.set('Vary', 'Origin')
+  }
+}
+
+function configureMethods(options: CorsOptions, req: Request, headers: Headers) {
+  if (options.methods && options.methods.length) {
+    headers.set('Access-Control-Allow-Methods', options.methods.join(','))
+  }
+}
+
+function configureCredentials(options: CorsOptions, req: Request, headers: Headers) {
+  if (options.credentials === true) {
+    headers.set('Access-Control-Allow-Credentials', 'true')
+  }
+}
+
+function configureAllowedHeaders(options: CorsOptions, req: Request, headers: Headers) {
+  let allowedHeaders: string[] | string | null = options.allowedHeaders || null
+
+  if (!allowedHeaders) {
+    allowedHeaders = req.headers.get('access-control-request-headers')! // .headers wasn't specified, so reflect the request headers
+    headers.set('Vary', 'Access-Control-Request-Headers')
+  }
+  if (allowedHeaders && !isString(allowedHeaders)) {
+    allowedHeaders = allowedHeaders.join(',') // .headers is an array, so turn it into a string
+  }
+  if (allowedHeaders && allowedHeaders.length) {
+    headers.set('Access-Control-Allow-Headers', allowedHeaders)
+  }
+}
+
+function configureExposedHeaders(options: CorsOptions, req: Request, headers: Headers) {
+  const exposedHeaders = options.exposedHeaders
+  if (exposedHeaders && exposedHeaders.length) {
+    headers.set('Access-Control-Expose-Headers', exposedHeaders.join(','))
+  }
+}
+
+function configureMaxAge(options: CorsOptions, req: Request, headers: Headers) {
+  if (options.maxAge !== undefined) {
+    headers.set('Access-Control-Max-Age', maxAge.toString())
```

Comment:
Ups, broke the build 🤦 
```suggestion
    headers.set('Access-Control-Max-Age', options.maxAge.toString())
```